pkg/workspace/templates_test.go

```diff
@@ -241,3 +241,43 @@ func TestRetrieveFileTemplate(t *testing.T) {
 		})
 	}
 }
+
+func TestProjectNames(t *testing.T) {
+	tests := []struct {
+		testName    string
+		projectName string
+		expectError bool
+	}{
+		{
+			testName:    "Correct Project Name",
+			projectName: "SampleProject",
+			expectError: false,
+		},
+		{
+			testName:    "Project Name with unsupported punctuation",
+			projectName: "SampleProject!",
+			expectError: true,
+		},
+		{
+			testName:    "Project Name starting with the word Pulumi",
+			projectName: "PulumiProject",
+			expectError: true,
+		},
+		{
+			testName:    "Project Name greater than 100 characters",
+			projectName: "cZClTe6xrjgKzH5QS8rFEPqYK1z4bbMeMr6n89n87djq9emSAlznQXXkkCEpBBCaZAFNlCvbfqVcqoifYlfPl11hvekIDjXVIY7m1",
+			expectError: true,
+		},
```

Comment:
Consider adding a test for the name `Pulumi`

Reply:
Done!